plugins/colorsource/export.js

```diff
@@ -0,0 +1,332 @@
+}
+
+/**
+ * @param {Mode} mode The personality's mode.
+ * @returns {Array.<Object>} Array of ColorSource commands (like "Reset"), may be empty. The commands are generated by Maintenance capabilities with hold time set.
+ */
+function getCommands(mode) {
+  const commands = [];
+  mode.channels.forEach((channel, channelIndex) => {
+    if (!channel.capabilities) {
+      // e. g. fine channels
+      return;
+    }
+
+    channel.capabilities.forEach(cap => {
+      if (cap.type === `Maintenance` && cap.hold) {
+        commands.push({
+          name: cap.comment,
+          steps: [
+            {
+              actions: [{
+                dmx: channelIndex,
+                value: cap.getMenuClickDmxValueWithResolution(CoarseChannel.RESOLUTION_8BIT)
+              }],
+              wait: 0 // this is apparently the delay before this step is activated
+            },
+            {
+              actions: [{
+                dmx: channelIndex,
+                value: -1
+              }],
+              wait: cap.hold.getBaseUnitEntity().number
+            }
+          ]
+        });
+      }
+    });
+  });
+
+  return commands;
+}
+
+/**
+ * @param {Mode} mode The personality's mode.
+ * @param {Boolean} hasIntensity Whether the mode has an intensity channel. Should equal fixtureJson.hasIntensity
+ * @returns {Array.<Object>} ColorSource channel objects of all mode channels.
+ */
+function getCSChannels(mode, hasIntensity) {
+  const channels = [];
+  mode.channels.forEach((channel, channelIndex) => {
+    const name = channel.name;
+
+    if (channel instanceof SwitchingChannel) {
+      channel = channel.defaultChannel;
+    }
+
+    const channelJson = {
+      coarse: channelIndex,
+      fadeWithIntensity: false,
+      fine: null,
+      highlight: 65535,
+      home: 0,
+      invert: false,
+      name,
+      ranges: [],
+      size: 8,
+      snap: false,
+      type: getCSChannelType(channel)
+    };
+
+    if (channelJson.type === CHANNEL_TYPE_COLOR) {
+      if (isMatrixChannel(channel)) {
+        channelJson.type = CHANNEL_TYPE_BEAM;
+      }
+      else if (channel.color) { // it may also be Hue or Saturation, which have no color
+        channelJson.name = channel.color.replace(/ /g, ``); // e.g. 'Warm White' -> 'WarmWhite'
+      }
+    }
+
+    if (channel instanceof FineChannel) {
+      if (channel.resolution === CoarseChannel.RESOLUTION_16BIT) {
+        // already handled by "fine" attribute of coarse channel
+        return;
+      }
+
+      channelJson.type = CHANNEL_TYPE_BEAM;
+      channelJson.home = scaleDmxValue(channel.defaultValue, CoarseChannel.RESOLUTION_8BIT, CoarseChannel.RESOLUTION_16BIT);
+    }
+    else {
+      addChannelDetails(channelJson, channel);
+    }
+
+    removeEmptyProperties(channelJson);
+
+    channels.push(channelJson);
+  });
+
+  return channels;
+
+  /**
+   * Adds detailed information to given channel JSON that only a CoarseChannel can provide.
```

Comment:
> […] that only a CoarseChannel can provide.

What else should provide it? Do you mean the following?

```suggestion
   * Adds detailed information to given channel JSON that can be deduced from only the corresponding CoarseChannel.
```